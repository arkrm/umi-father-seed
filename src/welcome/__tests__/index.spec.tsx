import React from 'react';
import { shallow, render } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Welcome, { prefixCls } from '../index';

describe('Welcome Component', () => {
  it('test render', () => {
    const wrapper = render(<Welcome ds={['Hello']} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('test props - btnText', () => {
    const btnText = '进入主页';
    const wrapper = shallow(<Welcome ds={['Hello']} btnText={btnText} />);
    expect(wrapper.find(`.${prefixCls}-link`).text()).toBe(btnText);
  });

  it('test props - onBtnClick', () => {
    const onClick = jest.fn();
    const wrapper = shallow(<Welcome ds={['Hello']} onBtnClick={onClick} />);
    wrapper.find(`.${prefixCls}-link`).simulate('click');
    expect(onClick).toBeCalled();
  });
});
