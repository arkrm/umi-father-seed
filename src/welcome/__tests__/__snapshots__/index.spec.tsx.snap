// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Component test render 1`] = `
<div
  class="ufs-welcome-wrapper"
>
  <div
    class="ufs-welcome-tag"
  >
    Hello
  </div>
  <div
    class="ufs-welcome-link"
  >
    ENTER
  </div>
</div>
`;
